### PR TITLE
Nuke caber

### PR DESCRIPTION
**Pull request by woisalreadytaken**
This change focuses on what the caber is supposed to be: a weapon to use when you have no other options.

For the simple part, all this really does is amplify what the caber already does: you deal more damage with it and never survive damage from it (and by never I really mean it), while adding fitting visual and sound effects to it. Explosion radius is unchanged.

A couple of things for the config were also changed to support this and maybe other things in the future:
- **Tags_Explode** now has custom `particle` and `sound` parameters, if the tag is used but these parameters aren't specified it'll use the same visual/sound effects it always has.
    - The `sound` parameter supports up to 16 non-custom sounds and will play one of them at random for the explosion.
- **Tags_ForceSuicide** has been added, it just kills the target.

### CODE REVIEW

File: addons/sourcemod/scripting/saxtonhale.sp [hunk 1]
```diff
@@ -32,6 +32,7 @@
 #define MAX_BUTTONS 					26
 #define MAX_TYPE_CHAR					32		//Max char size of methodmaps name
 #define MAXLEN_CONFIG_VALUE 			256		//Max config string buffer size
+#define MAXLEN_CONFIG_ALLVALUES			1024	//Max config buffer size for groups of values
```
Comment (FortyTwoFortyTwo) [inline addons/sourcemod/scripting/saxtonhale.sp]: `MAXLEN_CONFIG_VALUEARRAY`? May fit better as it should only be used on possible arrays

File: addons/sourcemod/configs/vsh/vsh.cfg [hunk 2]
```diff
@@ -1234,20 +1238,54 @@
...
+					"damagetype"	"blast"	//Only when the caber detonates
 				}
 				
 				"params"
 				{
-					"multiply"		"2.0"	//Multiply explosion damage by 2
+					"multiply"		"3.0"	//Multiply explosion damage by 3
+				}
+				
+				"Tags_Explode"
```
Comment (FortyTwoFortyTwo) [inline addons/sourcemod/configs/vsh/vsh.cfg]: Should add target param to victim

File: addons/sourcemod/scripting/vsh/tags/tags_params.sp [hunk 3]
```diff
@@ -50,11 +50,11 @@ methodmap TagsParams < StringMap
 	{
 		if (this == null) return false;
 		
-		char sValue[MAXLEN_CONFIG_VALUE];
+		char sValue[MAXLEN_CONFIG_ALLVALUES];
 		if (!this.GetString(sKey, sValue, sizeof(sValue)))
 			return false;
 		
-		char sValues[32][32];
+		char sValues[16][MAXLEN_CONFIG_VALUE];
```
Comment (FortyTwoFortyTwo) [inline addons/sourcemod/scripting/vsh/tags/tags_params.sp]: May well then add new define for `16`, `MAX_CONFIG_ARRAY`?

File: addons/sourcemod/configs/vsh/vsh.cfg [hunk 4]
```diff
@@ -1234,20 +1238,54 @@
...
+			"attrib"		"414 ; 1.0 ; 207 ; 11.0"
+			
+			"spawn"
+			{
+				"Tags_SetEntProp"	//Make the caber fucking massive
+				{
+					"target"		"melee"
+					"type"			"float"
+					"prop"			"m_flModelScale"
+					"value"			"3.0"
```
Comment (FortyTwoFortyTwo) [inline addons/sourcemod/configs/vsh/vsh.cfg]: I dunno, 3.0 may be a bit overkill but it just a visual change

File: addons/sourcemod/scripting/vsh/tags.sp [hunk 5]
```diff
@@ -783,11 +783,19 @@ public void Tags_Explode(int iClient, int iTarget, TagsParams tParams)
...
 	
+	//If no particle was specified, pick a generic explosion particle
+	char sParticle[MAXLEN_CONFIG_VALUE];
+	if (!tParams.GetString("particle", sParticle, sizeof(sParticle)))
+		Format(sParticle, sizeof(sParticle), "ExplosionCore_MidAir");
+	
+	//If no sounds were specified, pick a generic explosion sound. If multiple sounds were specified, pick a random one
+	char sSound[MAXLEN_CONFIG_VALUE];
+	if (!tParams.GetStringRandom("sound", sSound, sizeof(sSound)))
+		Format(sSound, sizeof(sSound), "weapons/airstrike_small_explosion_0%i.wav", GetRandomInt(1,3));
```
Comment (FortyTwoFortyTwo) [inline addons/sourcemod/scripting/vsh/tags.sp]: `%d` and spacing between `1,` and `3` random numbers

File: addons/sourcemod/configs/vsh/vsh.cfg (hunk 2 above)
Comment (woisalreadytaken) [reply addons/sourcemod/configs/vsh/vsh.cfg]: did this in a different way, made the filter consider melee and blast damage instead

**Pull request merged**